File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/loadmore/CustomLoadMoreView.java
package com.chad.baserecyclerviewadapterhelper.loadmore;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.chad.baserecyclerviewadapterhelper.R;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.chad.library.adapter.base.loadmore.BaseLoadMoreView;

import org.jetbrains.annotations.NotNull;

/**
 * Created by BlingBling on 2016/10/15.
 */

public final class CustomLoadMoreView extends BaseLoadMoreView {

    @NotNull
    @Override
    public View getRootView(@NotNull ViewGroup parent) {
        return LayoutInflater.from(parent.getContext()).inflate(R.layout.view_load_more, parent, false);
    }

    @NotNull
    @Override
    public View getLoadingView(@NotNull BaseViewHolder holder) {
        return holder.findView(R.id.load_more_loading_view);
    }

    @NotNull
    @Override
    public View getLoadComplete(@NotNull BaseViewHolder holder) {
        return holder.findView(R.id.load_more_load_complete_view);
    }

    @NotNull
    @Override
    public View getLoadEndView(@NotNull BaseViewHolder holder) {
        return holder.findView(R.id.load_more_load_end_view);
    }

    @NotNull
    @Override
    public View getLoadFailView(@NotNull BaseViewHolder holder) {
        return holder.findView(R.id.load_more_load_fail_view);
    }

}

File: kotlinapp/src/main/java/com/allen/kotlinapp/loadmore/CustomLoadMoreView.kt
package com.allen.kotlinapp.loadmore

import com.allen.kotlinapp.R
import com.chad.library.adapter.base.loadmore.LoadMoreView

/**
 * 文 件 名: CustomLoadMoreView
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 15:15
 * 修改时间：
 * 修改备注：
 */
class CustomLoadMoreView : LoadMoreView() {

    override fun getLayoutId(): Int {
        return R.layout.view_load_more
    }

    override fun getLoadingViewId(): Int {
        return R.id.load_more_loading_view
    }

    override fun getLoadFailViewId(): Int {
        return R.id.load_more_load_fail_view
    }

    override fun getLoadEndViewId(): Int {
        return R.id.load_more_load_end_view
    }
}
